contracts/strategies/RFPStrategy.sol

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import {BaseStrategy} from "./BaseStrategy.sol";
+import {Metadata} from "../core/libraries/Metadata.sol";
+import {ReentrancyGuard} from "@openzeppelin/security/ReentrancyGuard.sol";
+
+contract RFPStrategy is BaseStrategy, ReentrancyGuard {
+    /// @notice Struct to hold details of an recipient
+    struct Recipient {
+        address payoutAddress;
+        uint256 proposedAmount;
+        RecipientStatus recipientStatus;
+    }
+
+    struct Payout {
+        address recipient;
+        uint256 amount;
+    }
+
+    /// ===============================
+    /// ========== Errors =============
+    /// ===============================
+
+    error READY_FOR_PAYOUT();
+    error NOT_READY_FOR_PAYOUT();
+    error NOT_IMPLEMENTED();
+    error NOT_ELIGIBLE();
+    error UNAUTHORIZED();
+
+    /// ================================
+    /// ========== Storage =============
+    /// ================================
+
+    bool public payoutReady;
+    uint256 private _counter;
+    address public acceptedRecipientId;
+
+    /// @notice recipientId -> Recipient
+    mapping(address => Recipient) public recipients;
+
+    /// ======================
+    /// ======= Events =======
+    /// ======================
+
+    event RecipientRegistered(
+        address indexed recipientId, address payoutAddress, Metadata metadata, address indexed sender
+    );
+    event Allocated(bytes data, address indexed allocator);
+    event Distributed(address indexed recipientId, address recipient, uint256 amount, address token);
+
+    /// ===============================
+    /// ======== Constructor ==========
+    /// ===============================
+
+    constructor(address _allo, string memory _name) BaseStrategy(_allo, _name) {}
+
+    /// ====================================
+    /// =========== Functions ==============
+    /// ====================================
+
+    function initialize(uint256 _poolId, bytes memory _data) public override {
+        super.initialize(_poolId, _data);
+    }
+
+    /// @notice Submit proposal to RFP pool
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the transaction
+    function registerRecipients(bytes memory _data, address _sender) external payable onlyAllo returns (address) {
+        if (payoutReady) {
+            revert READY_FOR_PAYOUT();
+        }
+
+        (address payoutAddress, uint256 proposedAmount, Metadata memory metadata) =
+            abi.decode(_data, (address, uint256, Metadata));
+
+        Recipient memory recipient = Recipient({
+            payoutAddress: payoutAddress,
+            proposedAmount: proposedAmount,
+            recipientStatus: RecipientStatus.Pending
+        });
+
+        address recipientId = address(uint160(_counter++));
+        recipients[recipientId] = recipient;
```

Comment:
Is the reason to make this a counter so that the same address can submit twice? I thought we settled on recipientId being the address of the recipient, but this does seem to be a problem with that format.

Reply:
Updated this to be msg.sender / identityId from registry